### PR TITLE
ENH/API: overhaul simulated devices

### PR DESCRIPTION
**Pull request by tacaswell**
This replaces #682

### CODE REVIEW

File: ophyd/sim.py [hunk 1]
```diff
@@ -554,11 +555,12 @@ def stop(self, *, success=False):
...
 class NewTrivialFlyer(TrivialFlyer):
     """
-    The old-style API inserted Resource and Datum documents into a database directly.
-    The new-style API only caches the documents and provides an interface (collect_asset_docs)
-    for accessing that cache. This change was part of the "asset refactor" that changed
-    that way Resource and Datum documents flowed through ophyd, bluesky, and databroker.
-    Trivial flyer that complies to the API but returns empty data.
+    The old-style API inserted Resource and Datum documents into a database
+    directly. The new-style API only caches the documents and provides an
+    interface (collect_asset_docs for accessing that cache. This change was
```
Comment (jklynch) [inline ophyd/sim.py]: Lost ")" after "(collect_asset_docs"

File: ophyd/tests/test_epicsmotor.py [hunk 2]
```diff
@@ -4,60 +4,72 @@
...
 
-from ophyd import (EpicsMotor, Signal, EpicsSignal, EpicsSignalRO,
-                   Component as C, MotorBundle)
-from ophyd.utils.epics_pvs import (AlarmSeverity, AlarmStatus)
+from ophyd import (
+    EpicsMotor,
+    Signal,
+    EpicsSignal,
+    EpicsSignalRO,
+    Component as C,
```
Comment (jklynch) [inline ophyd/tests/test_epicsmotor.py]: "Component as Cpt" convention flouted

Comment (tacaswell) [reply ophyd/tests/test_epicsmotor.py]: not a new flouting (this is just wrapping).  Early on we went back and forth of `C` or `Cpt` was clearer.

**Comment by prjemian** [comment]
My opinion: `C` is so small, it is easily overlooked.  Even newbies will come across this code so be more clear than `C`.

File: ophyd/tests/test_epicsmotor.py (hunk 2 above)
Comment (danielballan) [reply ophyd/tests/test_epicsmotor.py]: Fair point that this is just wrapping, but in a bunch of places in `sim.py` you went through and changed `Component` to `Cpt`. Why?

**Comment by klauer** [comment]
`C`, as far as I recall, came from our original whiteboarding sessions and regrettably made it into actual code. `Cpt` or full `Component` all the way, in my opinion.

**Comment by jklynch** [comment]
The bluesky tests are passing with two small changes in PR #1258.
The suitcase-csv tests are passing with these changes to ophyd.

File: ophyd/scaler.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 from .ophydobj import Kind
 from .signal import (EpicsSignal, EpicsSignalRO)
 from .device import Device
-from .device import (Component as C, DynamicDeviceComponent as DDC,
+from .device import (Component as Cpt, DynamicDeviceComponent as DDC,
                      FormattedComponent as FC)
```
Comment (mrakitin) [inline ophyd/scaler.py]: There was the `FCpt` usage somewhere else in this PR. Should it also be changed here?

File: ophyd/mca.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 from .status import DeviceStatus
 from .signal import (Signal, EpicsSignal, EpicsSignalRO)
-from .device import (Device, Component as C, DynamicDeviceComponent as DDC,
+from .device import (Device, Component as Cpt, DynamicDeviceComponent as DDC,
```
[the remark this answers is not in the document]
Comment (jklynch) [reply ophyd/mca.py]: yes

File: ophyd/sim.py [hunk 5]
```diff
@@ -540,11 +661,12 @@ def stop(self, *, success=False):
...
```
[the remark this answers is not in the document]
Comment (jklynch) [reply ophyd/sim.py]: done

File: ophyd/sim.py [hunk 6]
```diff
@@ -1313,6 +1425,7 @@ def hw(save_path=None):
         motor_no_hints1=motor_no_hints1,
         motor_no_hints2=motor_no_hints2,
         bool_sig=bool_sig,
+        signal=signal
```
[the remark this answers is not in the document]
Comment (jklynch) [reply ophyd/sim.py]: done

File: ophyd/tests/test_pseudopos.py [hunk 7]
```diff
@@ -9,54 +9,57 @@
 from copy import copy
 
 from ophyd.epics_motor import EpicsMotor
-from ophyd.pseudopos import (PseudoPositioner, PseudoSingle)
+from ophyd.pseudopos import PseudoPositioner, PseudoSingle
 
-from ophyd import (Component as C)
+from ophyd import Component as C
```
[the remark this answers is not in the document]
Comment (jklynch) [reply ophyd/tests/test_pseudopos.py]: yes

File: .travis.yml [hunk 8]
```diff
@@ -105,9 +105,16 @@ before_install:
...
 
 install:
-  - python -mpip install -Ur requirements.txt
-  - python -mpip install -Ur test-requirements.txt
+  # use a virtual environment  #- python -mpip install -Ur requirements.txt
+  #- python -mpip install -Ur test-requirements.txt
+
+  - python -m venv ophyd-venv
+  - source ophyd-venv/bin/activate
+  - python -mpip install --upgrade pip
```
Comment (mrakitin) [inline .travis.yml]: Can we have a blank space between `-m` and `pip` to make it more readable here and 2 lines below?

**Comment by mrakitin** [comment]
Looks like the base version of Python for venv on Travis is 2.7, which causes all failures.

File: .travis.yml (hunk 8 above)
Comment (jklynch) [reply .travis.yml]: Yes.

Comment (jklynch) [reply .travis.yml]: Also, Travis already uses a virtual environment so I was just making trouble with that.

**Comment by mrakitin** [comment]
Thanks for pushing on this PR, @jklynch! The tests are passing now. Let's give our west-coast colleagues a chance to review it before it gets merged.

**Pull request merged**